searx/version.py

```diff
@@ -59,7 +59,7 @@ def get_git_url_and_branch():
 
 
 def get_git_version():
-    git_commit_date_hash = subprocess_run(r"git show -s --date='format:%Y.%m.%d' --format='%cd+%h'")
+    git_commit_date_hash = subprocess_run(r"git show -s --date='format:%-Y.%-m.%-d' --format='%cd+%h'")
```

Comment:
`YYYY-MM-DD` is not a valid **number** for a (major) release in sense of semantic versioning --> `YYYY-MM-DD` is  a string --> only **numbers** (integers) can be selected by version specifiers

- https://peps.python.org/pep-0440/#version-specifiers
- https://peps.python.org/pep-0440/#public-version-identifiers

Reply:
The `-` removes the leading zero

Reply:
Oops sorry. :facepalm: